src/lib/djerba/plugins/fusion/plugin.py

```diff
@@ -63,12 +67,129 @@ def sort_by_actionable_level(row):
+                self.logger.error(f"Error processing fusion {fusion}: {e}")
+
+        # Save the fusion-URL pairs to a CSV file
+        output_tsv_path = os.path.join(output_dir, 'fusion_blurb_urls.tsv')
+        with open(output_tsv_path, 'w', newline='') as tsvfile:
+            writer = csv.writer(tsvfile, delimiter='\t')
+            writer.writerow(['Fusion', 'Whizbam URL'])
+            writer.writerows(fusion_url_pairs)
+
         data = self.get_starting_plugin_data(wrapper, self.PLUGIN_VERSION)
         data[core_constants.RESULTS] = results
         data[core_constants.MERGE_INPUTS]['gene_information_merger'] = gene_info
         data[core_constants.MERGE_INPUTS]['treatment_options_merger'] = treatment_opts
         return data
 
+    def process_fusion(self, config, fusion, tsv_file_path, json_template_path, output_dir):
+        base_dir = (directory_finder(self.log_level, self.log_path).get_base_dir())
+        wrapper = self.get_config_wrapper(config)
+        match = re.match(r"(.+)::(.+)", fusion)
+        if match:
+            gene1 = match.group(1)
+            gene2 = match.group(2)
+        else:
+            raise ValueError(f"No valid fusion found for {fusion}.")
+
+        # Initialize breakpoints
+        breakpoint1, breakpoint2 = None, None
+
+        # Open and read the ARRIBA TSV file
+        with open(tsv_file_path, mode='r') as file:
+            reader = csv.DictReader(file, delimiter='\t')
+
+            # Find the correct row based on gene1 and gene2
+            for row in reader:
+                if (row['#gene1'] == gene1 or row['#gene1'] == gene2) and (
+                        row['gene2'] == gene1 or row['gene2'] == gene2):
+                    breakpoint1 = row['breakpoint1']
+                    breakpoint2 = row['breakpoint2']
+                    break
+
+        if not (breakpoint1 and breakpoint2):
+            raise ValueError(f"No matching fusion found in the TSV file for {fusion}.")
+
+        # Function to modify the breakpoint format to "chr:start-end"
+        def format_breakpoint(breakpoint):
+            chrom, pos = breakpoint.split(':')
+            start = int(pos)
+            end = start + 1
+            return f"{chrom}:{start}-{end}"
+
+        formatted_breakpoint1 = format_breakpoint(breakpoint1)
+        formatted_breakpoint2 = format_breakpoint(breakpoint2)
+
+        # Load the JSON template
+        with open(json_template_path, 'r') as json_file:
+            data = json.load(json_file)
+
+        # Update the JSON with the formatted breakpoints
+        data['locus'] = [formatted_breakpoint1, formatted_breakpoint2]
+
+        tumour_id = wrapper.get_my_string(core_constants.TUMOUR_ID)
+        data['tracks'][1]['name'] = tumour_id
+
+        # Search for the BAM and BAI files using glob.glob
+        bam_pattern = f"/.mounts/labs/prod/whizbam/*/RNASEQ/{tumour_id}.bam"
+        bai_pattern = f"/.mounts/labs/prod/whizbam/*/RNASEQ/{tumour_id}.bai"
```

Comment:
Not good to have these near-identical patterns hard-coded -- at the least we should define a constant, something like `WHIZBAM_PATTERN_ROOT=/.mounts/labs/prod/whizbam/*/RNASEQ/`. We could also consider having a config file for paths/patterns.